src/main/java/graphql/annotations/connection/ConnectionTypeValidator.java

```diff
@@ -0,0 +1,42 @@
+/**
+ * Copyright 2016 Yurii Rashkovskii
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ */
+package graphql.annotations.connection;
+
+import graphql.annotations.GraphQLDataFetcher;
+
+import java.lang.reflect.AccessibleObject;
+import java.lang.reflect.Field;
+import java.lang.reflect.Method;
+
+public class ConnectionTypeValidator {
+
+    public void validate(AccessibleObject field) {
+        if (field instanceof Field) {
+            if (field.isAnnotationPresent(GraphQLConnection.class) && !field.isAnnotationPresent(GraphQLDataFetcher.class)) {
+                throw new GraphQLConnectionException("Please don't use @GraphQLConnection on" + ((Field) field).getName() +
+                        " without @GraphQLDataFetcher, because " +
+                        "neither PropertyDataFetcher nor FieldDataFetcher know how to handle connection");
+            }
+
+            if (!PaginatedData.class.isAssignableFrom(((Field) field).getType())) {
```

Comment:
From my understanding, PaginatedData is only required for "EnhancedConnectionFetcher", but we could perfectly imagine that somebody implement another ConnectionFetcher not based on PaginatedData. Actually if someone really wants to have the same kind of stuff as the previous SimpleListConnection based on a List, the validator should not block it. I think the validator class should actually be provided by the EnhancedConnectionFetcher itself (either through an annotation on EnhancedConnectionFetcher, or through a static method ?)

Reply:
now this class implements an interface called ConnectionValidator, and it can be overridden
in EnhancedConnectionFetcher (now called PaginatedDataConnectionFetcher)